---
title: "Gene list summary"
author: "Amy Hicks"
date: "03/02/21"
output:
  html_document:
    code_folding: hide
    toc: true
    toc_float: true
    df_print: paged
---
```{r setup, include=FALSE}
library(tidyverse)
library(biomaRt)
library(readxl)
library(knitr)
library(here)
knitr::opts_chunk$set(echo = TRUE)
```

Gene lists were collated from a number of different sources and processed in preparation for exploration of gene co-expression networks. Files containing lists of `gene_symbols` and `ENSGs` were saved for future analyses. 

# 1. NSL complex genes
Obtained from [Sheikh et al., 2019](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6607013/), this list contains genes encoding the nine proteins making up the non-specific lethal (NSL) complex which is a histone acetyltransferase complex linked to both nuclear and mitochondrial transcriptional regulation.
```{r}
# Import lists of genes of interest
NSL_file <- read.csv(file.path(here::here("data", "Gene_lists", "NSL_genes.csv")))
names(NSL_file)[names(NSL_file)=="Symbol"] <- "Gene_symbol"
# Open ensembl
ensembl = useMart("ensembl", dataset = "hsapiens_gene_ensembl")
# Find corresponding ENSGs
NSL_ENSGs_list <- getBM(attributes = c("hgnc_symbol", "ccds", "ensembl_gene_id"), filters = "hgnc_symbol", values = NSL_file, mart = ensembl)
# Retain only primary assembly ENSGs
NSL_list <- NSL_ENSGs_list[!(NSL_ENSGs_list$ccds==""), ] %>%
  dplyr::distinct(ensembl_gene_id, .keep_all = TRUE) %>%
  dplyr::select(hgnc_symbol, ensembl_gene_id)
names(NSL_list)[names(NSL_list)=="hgnc_symbol"] <- "Gene_symbol"
names(NSL_list)[names(NSL_list)=="ensembl_gene_id"] <- "Ensembl_ID"
```

The total number of genes in this list: **`r nrow(NSL_list)`**. 
```{r}
# Print list
dplyr::arrange(NSL_list, Gene_symbol)
```

# 2. Mendelian Parkinson's disease genes 
The final list used for analysis was a broad list of genes relating unexclusively to mendelian PD, obtained from PanelApp [Parkinson disease and complex Parkinsonism version 1.68](https://panelapp.genomicsengland.co.uk/panels/39/), a list collected and validated by a group of experts.
```{r} 
# Upload file
PD_mendelian_file <- read_excel(file.path(here::here("data", "Gene_lists", "PD_and_complex_Parkinsonism.xlsx")))
# Extract columns of interest
PD_mendelian_list <- data.frame(PD_mendelian_file$`Gene Symbol`, PD_mendelian_file$`EnsemblId(GRch37)`)
names(PD_mendelian_list)[names(PD_mendelian_list)=="PD_mendelian_file..Gene.Symbol."] <- "Gene_symbol"
names(PD_mendelian_list)[names(PD_mendelian_list)=="PD_mendelian_file..EnsemblId.GRch37.."] <- "Ensembl_ID"
PD_mendelian_list <- dplyr::arrange(PD_mendelian_list)
```

The total number of genes in this list: **`r nrow(PD_mendelian_list)`**.
```{r}
# Print list
PD_mendelian_list
```

# 3. Sporadic Parkinson's disease genes
The final list used for analysis was a broad list mentioned in [Nalls et al., 2019](https://www.thelancet.com/journals/laneur/article/PIIS1474-4422(19)30320-5/fulltext), described as *"Of the genes possibly associated with at least one QTL in public reference datasets and  therefore testable via summary-based Mendelian randomisation, the expression or methylation of 64% was significantly associated with a possible causal change in Parkinson's disease risk."*
```{r}
# Upload file
PD_GWAS_file <- read_excel(file.path(here::here("data", "Gene_lists", "GWAS_Nalls_2019_mendrand.xlsx")))
# Extract columns of interest
PD_GWAS_list <- data.frame(PD_GWAS_file$Probe, PD_GWAS_file$Gene, PD_GWAS_file$`Top SNP`, PD_GWAS_file$`Pass Bonferroni`)
names(PD_GWAS_list)[names(PD_GWAS_list)=="PD_GWAS_file.Gene"] <- "Gene"
names(PD_GWAS_list)[names(PD_GWAS_list)=="PD_GWAS_file..Pass.Bonferroni."] <- "Pass_Bonferroni"
# Remove genes that failed final step
PD_GWAS_list <- dplyr::filter(PD_GWAS_list, Pass_Bonferroni == "pass")
# Remove duplicates, create gene list
PD_GWAS_genes <- dplyr::distinct(PD_GWAS_list, Gene)
# Obtain ENSG numbers for each gene
PD_GWAS_ENSG <- getBM(attributes = c("hgnc_symbol", "ccds", "ensembl_gene_id"), filters = "hgnc_symbol", values = PD_GWAS_genes, mart = ensembl)
# Retain only primary assembly ENSGs
PD_GWAS_list <- PD_GWAS_ENSG[!(PD_GWAS_ENSG$ccds==""), ] %>%
  dplyr::distinct(ensembl_gene_id, .keep_all = TRUE) %>%
  dplyr::select(hgnc_symbol, ensembl_gene_id) %>%
  dplyr::arrange(hgnc_symbol)
```

The total number of genes in this list: **`r nrow(PD_GWAS_genes)`**. The number of corresponding ENSGs found is **`r nrow(PD_GWAS_list)`**, meaning there were `r nrow(PD_GWAS_genes) - nrow(PD_GWAS_list)` missing.
```{r}
# Find missing ENSGs
GWAS_missing <- dplyr::setdiff(PD_GWAS_genes$Gene, PD_GWAS_list$hgnc_symbol)
GWAS_missing_ENSG <- getBM(attributes = c("hgnc_symbol", "ensembl_gene_id"), filters = "hgnc_symbol", values = GWAS_missing, mart = ensembl)
# Remove rogue extra ENSG for one gene
GWAS_missing_ENSG <- GWAS_missing_ENSG %>%
  dplyr::filter(ensembl_gene_id != "ENSG00000284832")
# Manually search genes still missing ENSGs
GWAS_missing2 <- dplyr::setdiff(GWAS_missing, GWAS_missing_ENSG$hgnc_symbol)
GWAS_missing2 <- as.data.frame(GWAS_missing2[2:4])
GWAS_missing2$ensembl_gene_id <- c("ENSG00000197238", "ENSG00000225914", "ENSG00000136243")
names(GWAS_missing2)[names(GWAS_missing2)=="GWAS_missing2[2:4]"] <- "hgnc_symbol"
# Add onto table
PD_GWAS_list <- dplyr::bind_rows(PD_GWAS_list, GWAS_missing_ENSG, GWAS_missing2)
names(PD_GWAS_list)[names(PD_GWAS_list)=="hgnc_symbol"] <- "Gene_symbol"
names(PD_GWAS_list)[names(PD_GWAS_list)=="ensembl_gene_id"] <- "Ensembl_ID"
```

The number of ENSGs in this list is now: **`r nrow(PD_GWAS_list)`**.
```{r}
# Show which is missing
setdiff(PD_GWAS_genes$Gene, PD_GWAS_list$Gene_symbol)
```

This remaining missing gene is a **non-coding RNA** and does not have an associated ENSG. 
```{r}
# Print list
PD_GWAS_list
```
