---
title: "Sec_GCN_ukbec"
author: "Amy Hicks"
date: "16/12/2021"
output:
  html_document:
    code_folding: hide
    toc: true
    toc_float: true
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(ggplot2)
library(devtools)
library(CoExpNets)
library(CoExp10UKBEC)
library(stringr)
library(corrplot)
library(biomaRt)
library(knitr)
library(readxl)
library(gprofiler2)
library(rmarkdown)
library(rutils)
library(GMSCA)
library(here)
```
This markdown contains an exploration of the secondary coexpression networks from [Sanchez et al., 2021](https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btab175/6178279). Four networks were created from frontal cortex tissue data from the **UKBEC dataset** and categorised by the correction or **removal of the effect** of specific cell types, namely **neurons, mature astrocytes, microglia** and **oligodendrocytes**. Several lists of genes of interest were searched within these networks to investigate the effect of cell type on gene co-expression relationships.

```{r, results = FALSE}
# First initialise the datasets/packages to use
CoExp10UKBEC::initDb(mandatory = T)
```

# 1. Gene lists  of interest
## 1.1. NSL complex
Firstly, the nine genes encoding the NSL complex will be focused on [(Sheikh et al., 2019)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6607013/). These genes encode proteins which make up an **acetyltransferase complex** linked to both nuclear and mitochondrial gene expression.
```{r}
# Load list
NSL_list <- read.delim(file.path(here::here("data", "Gene_lists", "NSL_list.txt")))
# Add column with gene symbols known to be in UKBEC dataset
NSL_list$UKBEC <- NSL_list$Gene_symbol
NSL_list$UKBEC[NSL_list$UKBEC=="KANSL1"] <- "KIAA1267"
NSL_list$UKBEC[NSL_list$UKBEC=="KAT8"] <- "MYST1"
NSL_list$UKBEC[NSL_list$UKBEC=="KANSL2"] <- "C12orf41"
NSL_list$UKBEC[NSL_list$UKBEC=="KANSL3"] <- "KIAA1310"
# Edit add column with combination of UKBEC symbols and ENSGs
NSL_list$UKBEC_symbol <- as.character(NSL_list$Ensembl_ID)
NSL_list$UKBEC_symbol[NSL_list$UKBEC_symbol=="ENSG00000120071"] <- "KIAA1267"
NSL_list$UKBEC_symbol[NSL_list$UKBEC_symbol=="ENSG00000103510"] <- "MYST1"
NSL_list$UKBEC_symbol[NSL_list$UKBEC_symbol=="ENSG00000139620"] <- "C12orf41"
NSL_list$UKBEC_symbol[NSL_list$UKBEC_symbol=="ENSG00000114982"] <- "KIAA1310"
# Make search list
NSL_ENSG <- as.character(NSL_list$UKBEC_symbol)
```

The number of genes in this list: **`r nrow(NSL_list)`**.

## 1.2. PD mendelian
A list of highly penetrant genes linked to familial forms of Parkinson's disease was obtained from PanelApp [Parkinson disease and complex Parkinsonism version 1.68](https://panelapp.genomicsengland.co.uk/panels/39/), a list collected and validated by a group of experts.
```{r} 
# Load list
PD_mendelian_list <- read.delim(file.path(here::here("data", "Gene_lists", "PD_mendelian_list.txt")))
# Add column with gene symbols known to be in UKBEC dataset
PD_mendelian_list$UKBEC <- as.character(PD_mendelian_list$Gene_symbol)
PD_mendelian_list$UKBEC[PD_mendelian_list$UKBEC=="TUBB4A"] <- "TUBB4"
# Edit add column with combination of UKBEC symbols and ENSGs
PD_mendelian_list$UKBEC_symbol <- as.character(PD_mendelian_list$Ensembl_ID)
PD_mendelian_list$UKBEC_symbol[PD_mendelian_list$UKBEC_symbol=="ENSG00000104833"] <- "TUBB4"
# Make search list
PD_mendelian_ENSG <- as.character(PD_mendelian_list$UKBEC_symbol)
```

The number of genes in this list: **`r nrow(PD_mendelian_list)`**.

## 1.3. PD sporadic
A list of less penetrant, more common genes linked to sporadic Parkinson's disease  was obtained from the [latest GWAS](https://www.thelancet.com/journals/laneur/article/PIIS1474-4422(19)30320-5/fulltext), described as *"Of the genes possibly associated with at least one QTL in public reference datasets and  therefore testable via summary-based Mendelian randomisation, the expression or methylation of 64% was significantly associated with a possible causal change in Parkinson's disease risk."*
```{r}
# Load list
PD_GWAS_list <- read.delim(file.path(here::here("data", "Gene_lists", "PD_GWAS_list.txt")))
# Add column with gene symbols known to be in UKBEC dataset
PD_GWAS_list$UKBEC <- as.character(PD_GWAS_list$Gene_symbol)
#PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="CCAR2"] <- "KIAA1967"
PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="LINC00174"] <- "NCRNA00174"
PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="MALSU1"] <- "C7orf30"
#PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="PGF"] <- "PIGF"
PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="PPIP5K2"] <- "HISPPD1"
#PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="RAB29"] <- "RAB7L1"
#PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="RETREG3"] <- "FAM134C"
#PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="SGF29"] <- "CCDC101"
PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="TMEM248"] <- "C7orf42"
PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="ZKSCAN8"] <- "ZNF192"
PD_GWAS_list$UKBEC[PD_GWAS_list$UKBEC=="ZNF192P1"] <- "ZNF389"
# Edit add column with combination of UKBEC symbols and ENSGs
PD_GWAS_list$UKBEC_symbol <- as.character(PD_GWAS_list$Ensembl_ID)
#PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000158941"] <- "KIAA1967"
PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000179406"] <- "NCRNA00174"
PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000156928"] <- "C7orf30"
#PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000119630"] <- "PIGF"
PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000145725"] <- "HISPPD1"
#PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000117280"] <- "RAB7L1"
#PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000141699"] <- "FAM134C"
#PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000176476"] <- "CCDC101"
PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000106609"] <- "C7orf42"
PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000198315"] <- "ZNF192"
PD_GWAS_list$UKBEC_symbol[PD_GWAS_list$UKBEC_symbol=="ENSG00000226314"] <- "ZNF389"
# Make search list
PD_GWAS_ENSG <- as.character(PD_GWAS_list$UKBEC_symbol)
```  

The number of genes in this list: **`r nrow(PD_GWAS_list)`**.

# 2. Useful info from primary network
Some information from the primary gene co-expression network was compared to results from secondary gene co-expression networks. 
```{r, results = FALSE, warning = FALSE}
# Load enrichments from primary network
ukbec_net <- readRDS(file.path(here::here("data", "CoExp_source", "netFCTX.12.signed.it.20.rds_cor_pca.rds")))
ukbec_enrich <- as.data.frame(getModulesEnrichment(ukbec_net, markers.path = file.path(here::here("data", "GMSCA_source", "markers"))))
# Search NSL genes in primary network
NSL_report_uk <- reportOnGenes(tissue = "FCTX", genes = NSL_ENSG, which.one = "10UKBEC")
NSL_rep_uk <- NSL_report_uk$report 
NSL_mods <- as.character(unique(NSL_rep_uk$module))
ukbec_enrich_NSL <- dplyr::select(ukbec_enrich, one_of(NSL_mods))
```

# 3. Neuron network
The overall structure of the network was first examined. 
```{r}
# Load network
N_net = readRDS(file.path(here::here("data", "Networks", "netFCTX.12.signed.it.20.rds.No.N.rds")))
# Plot sizes of modules making up the network
CoExpNets::plotModSizes(tissue = N_net, which.one = "new")
```

```{r}
# Obtain eigengenes for each module
N_net.name <- file.path(here::here("data", "Networks", "netFCTX.12.signed.it.20.rds.No.N.rds"))
N_egs <- CoExpNets::getNetworkEigengenes(tissue = N_net.name, which.one = "new")
# Plot eigengene clustering 
CoExpNets::plotEGClustering(tissue = N_net.name, which.one = "new")
```

```{r}
# Plot correlation between modules
corrplot(cor(N_egs), tl.srt = 45,
         tl.col = "black",
         type = "upper", 
         tl.cex = 0.45,
         order = "hclust",
         title = paste0("Eigengene correlations for UKBEC modules"),
         mar = c(3, 2, 2.5, 4))
```

## 3.1. Searching gene list {.tabset}
### NSL genes
First the genes encoding the NSL complex were searched. 
```{r}
# Search genes in network
N_NSL_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = N_net.name, genes = NSL_ENSG, which.one = "new", net = NULL))
N_NSL_mod <- tibble::rownames_to_column(N_NSL_mod, "ENSG")
N_NSL_mod <- dplyr::select(N_NSL_mod, ENSG, module, p.val.mods)
N_NSL_mod$Gene <- N_NSL_mod$ENSG
# Add gene symbols
for (i in NSL_list$UKBEC_symbol) {
  loc <- match(i, NSL_list$UKBEC_symbol)
  N_NSL_mod$Gene[N_NSL_mod$Gene== i] <- NSL_list$Gene_symbol[loc]
}
# Get module membership
N_MM <- as.data.frame(N_net$MM)
colnames(N_MM)<- "MM"
N_MM <- rownames_to_column(N_MM, "ENSG")
N_MM$MM <- signif(N_MM$MM, 4)
# Add MM to table
N_NSL_mod$MM <- N_NSL_mod$ENSG
for (i in N_MM$ENSG) {
  loc <- match(i, N_MM$ENSG)
  N_NSL_mod$MM[N_NSL_mod$MM== i] <- N_MM$MM[loc]
}
# Reorder table
N_NSL_mod <- N_NSL_mod[,c(4,1,5,2,3)]
N_NSL_mod <- dplyr::arrange(N_NSL_mod, p.val.mods)
# Store NSL enriched modules
N_NSL_modules <- as.character(unique(N_NSL_mod$module))
N_NSL_modules <- N_NSL_modules[N_NSL_modules!="void"]
# Print result
N_NSL_mod
```
```{r}
# Remove void
N_NSL_mod <- dplyr::filter(N_NSL_mod, module !="void")
# Format dataframe for figure
N_NSL_mod$FDR <- p.adjust(N_NSL_mod$p.val.mods, method = "fdr")
names(N_NSL_mod)[names(N_NSL_mod)=="p.val.mods"] <- "p.val"
names(N_NSL_mod)[names(N_NSL_mod)=="module"] <- "Module"
N_NSL_mod$p.val <- as.numeric(N_NSL_mod$p.val)
N_NSL_mod <- dplyr::arrange(N_NSL_mod, Module)
N_NSL_mod$`-log10(p.val)` <- -log10(N_NSL_mod$p.val)
N_NSL_mod$`-log10(FDR)` <- -log10(N_NSL_mod$FDR)
# Produce plot
source(file.path(here::here("scripts", "CoExp_functions.R")))
ggplot(N_NSL_mod, aes(Gene, Module)) +
  geom_tile(aes(fill = `-log10(FDR)`)) +
  geom_text(aes(label = signif.num(FDR))) +
  scale_fill_gradient(low = "white", high = "goldenrod3", limits = c(0,4)) +
  labs(y = "Module", x = "Gene")
```

The total number of genes in the output table was **`r nrow(N_NSL_mod)`**, indicating there were **`r nrow(NSL_list) - nrow(N_NSL_mod)`** genes missing from the search. These genes appeared in `r length(unique(N_NSL_modules))` modules, each of which was examined for **enrichment** of cell type specific markers. 
```{r, results = FALSE}
# Load cell type markers enriched for each module
N_mod_enrich <- as.data.frame(getModulesEnrichment(N_net, markers.path = file.path(here::here("data", "GMSCA_source", "markers"))))
# Filter enrichments for NSL modules
N_mod_enrich_NSL <- dplyr::select(N_mod_enrich, one_of(N_NSL_modules))
```

The modules containing NSL genes showed **no significant enrichment** for **any cell types**. 

### PD mendelian genes
The results tables for the mendelian PD gene list were filtered for the `r length(N_NSL_modules)` modules containing NSL genes. 
```{r}
# Search genes in network
N_PD_mendelian_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = N_net.name, genes = PD_mendelian_ENSG, which.one = "new", net = NULL))
N_PD_mendelian_mod <- tibble::rownames_to_column(N_PD_mendelian_mod, "ENSG")
N_PD_mendelian_mod <- dplyr::select(N_PD_mendelian_mod, ENSG, module, p.val.mods)
N_PD_mendelian_mod$Gene <- N_PD_mendelian_mod$ENSG
# Add gene symbols
for (i in PD_mendelian_list$UKBEC_symbol) {
  loc <- match(i, PD_mendelian_list$UKBEC_symbol)
  N_PD_mendelian_mod$Gene[N_PD_mendelian_mod$Gene== i] <- PD_mendelian_list$Gene_symbol[loc]
}
# Add MM to table
N_PD_mendelian_mod$MM <- N_PD_mendelian_mod$ENSG
for (i in N_MM$ENSG) {
  loc <- match(i, N_MM$ENSG)
  N_PD_mendelian_mod$MM[N_PD_mendelian_mod$MM== i] <- N_MM$MM[loc]
}
# Reorder table
N_PD_mendelian_mod <- N_PD_mendelian_mod[,c(4,1,5,2,3)]
N_PD_mendelian_mod <- dplyr::arrange(N_PD_mendelian_mod, p.val.mods)
# Filter for NSL enriched modules 
N_PD_mendelian_mod1 <- filter(N_PD_mendelian_mod, module %in% N_NSL_modules)
# Print result
N_PD_mendelian_mod1
```

The number of genes in the output table was **`r nrow(N_PD_mendelian_mod)`**, indicating there were **`r length(PD_mendelian_ENSG) - nrow(N_PD_mendelian_mod)`** genes missing from the search. 

### PD sporadic genes
The results tables for the sporadic PD gene list were filtered for the `r length(N_NSL_modules)` modules containing NSL genes. 
```{r}
# Search genes in network
N_PD_GWAS_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = N_net.name, genes = PD_GWAS_ENSG, which.one = "new", net = NULL))
N_PD_GWAS_mod <- tibble::rownames_to_column(N_PD_GWAS_mod, "ENSG")
N_PD_GWAS_mod <- dplyr::select(N_PD_GWAS_mod, ENSG, module, p.val.mods)
N_PD_GWAS_mod$Gene <- N_PD_GWAS_mod$ENSG
# Add gene symbols
for (i in PD_GWAS_list$Ensembl_ID) {
  loc <- match(i, PD_GWAS_list$Ensembl_ID)
  N_PD_GWAS_mod$Gene[N_PD_GWAS_mod$Gene== i] <- PD_GWAS_list$Gene_symbol[loc]
}
# Add MM to table
N_PD_GWAS_mod$MM <- N_PD_GWAS_mod$ENSG
for (i in N_MM$ENSG) {
  loc <- match(i, N_MM$ENSG)
  N_PD_GWAS_mod$MM[N_PD_GWAS_mod$MM== i] <- N_MM$MM[loc]
}
# Reorder table
N_PD_GWAS_mod <- N_PD_GWAS_mod[,c(4,1,5,2,3)]
N_PD_GWAS_mod <- dplyr::arrange(N_PD_GWAS_mod, p.val.mods)
# Filter for NSL enriched modules 
N_PD_GWAS_mod1 <- filter(N_PD_GWAS_mod, module %in% N_NSL_modules)
# Print result
N_PD_GWAS_mod1
```

The number of genes in the output table was **`r nrow(N_PD_GWAS_mod)`**, indicating there were **`r length(PD_GWAS_ENSG) - nrow(N_PD_GWAS_mod)`** genes missing from the search. 

## 3.2. Summary
The **frequency** of **genes** within each gene **list** apppearing in each of the **`r length(N_NSL_modules)` NSL enriched modules** was calculated. 
```{r}
# Collate module results into one table
N1 <- N_PD_mendelian_mod1['module']
N1['Gene_list'] <- "PD_mendelian"
N2 <- N_PD_GWAS_mod1['module']
N2['Gene_list'] <- "PD_GWAS"
N_mods_list <- do.call(rbind, list(N1, N2))
# Display as frequency of each module within each list
N_mod_freq <- with(N_mods_list, table(module, Gene_list))
N_mod_freq <- as.data.frame.matrix(N_mod_freq)
N_mod_freq <- dplyr::filter(N_mod_freq, rownames(N_mod_freq) %in% N_NSL_modules)
N_mod_freq <- N_mod_freq[c(2, 3, 4, 1),]
```

```{r}
knitr::kable(N_mod_freq)
```

The **significance** of the **enrichments** of each **gene list** within each of the **`r length(N_NSL_modules)` NSL enriched modules** was also calculated. P-values were **FDR-corrected** before summarising. 
```{r}
# FDR correct p values
N_PD_mendelian_mod1$FDR <- p.adjust(N_PD_mendelian_mod1$p.val.mods, method = "fdr")
N_PD_GWAS_mod1$FDR <- p.adjust(N_PD_GWAS_mod1$p.val.mods, method = "fdr")
```

```{r, include = FALSE}
# Collate module results into one table
Ns1 <- data.frame(N_PD_mendelian_mod1$module, N_PD_mendelian_mod1$FDR)
names(Ns1)[names(Ns1)=="N_PD_mendelian_mod1.FDR"] <- "FDR"
names(Ns1)[names(Ns1)=="N_PD_mendelian_mod1.module"] <- "Module"
Ns1$Gene_list <- "PD_mendelian"
Ns2 <- data.frame(N_PD_GWAS_mod1$module, N_PD_GWAS_mod1$FDR)
names(Ns2)[names(Ns2)=="N_PD_GWAS_mod1.FDR"] <- "FDR"
names(Ns2)[names(Ns2)=="N_PD_GWAS_mod1.module"] <- "Module"
Ns2$Gene_list <- "PD_GWAS"
N_pvals_list <- dplyr::bind_rows(Ns1, Ns2)
# Display as true/false table
N_TF <- N_pvals_list %>% 
  dplyr::distinct() %>%
  tidyr::spread(Gene_list, FDR)
N_TF <- tibble::column_to_rownames(N_TF, var = "Module")
N_TF <- N_TF[c(2, 3, 4, 1),]
```

```{r}
knitr::kable(N_TF)
```

```{r}
# Filter repeats
N_melt_summ <- N_pvals_list %>%
  dplyr::distinct()
# Prep for figure 
N_melt_summ$Module <- as.character(N_melt_summ$Module)
N_melt_summ$Gene_list <- as.character(N_melt_summ$Gene_list)
N_melt_summ$FDR <- as.numeric(N_melt_summ$FDR)
N_melt_summ$`-log10(FDR)` <- -log10(N_melt_summ$FDR)
N_melt_summ <- dplyr::arrange(N_melt_summ, Module)
# Plot
ggplot(N_melt_summ, aes(Gene_list, Module)) +
  geom_tile(aes(fill = `-log10(FDR)`)) +
  geom_text(aes(label = signif.num(FDR))) +
  scale_fill_gradient(low = "white", high = "lightsteelblue4", limits = c(0,4)) +
  labs(y = "Module", x = "Gene list")
```

# 4. Astrocyte network
The overall structure of the network was first examined. 
```{r}
# Load network
MA_net = readRDS(file.path(here::here("data", "Networks", "netFCTX.12.signed.it.20.rds.No.MA.rds")))
# Plot sizes of modules making up the network
CoExpNets::plotModSizes(tissue = MA_net, which.one = "new")
```

```{r}
# Obtain eigengenes for each module
MA_net.name <- file.path(here::here("data", "Networks", "netFCTX.12.signed.it.20.rds.No.MA.rds"))
MA_egs <- CoExpNets::getNetworkEigengenes(tissue = MA_net.name, which.one = "new")
# Plot eigengene clustering 
CoExpNets::plotEGClustering(tissue = MA_net.name, which.one = "new")
```

```{r}
# Plot correlation between modules
corrplot(cor(MA_egs), tl.srt = 45,
         tl.col = "black",
         type = "upper", 
         tl.cex = 0.45,
         order = "hclust",
         title = paste0("Eigengene correlations"),
         mar = c(3, 2, 2.5, 4))
```

## 4.1. Searching gene lists {.tabset}
### NSL genes
First the genes encoding the NSL complex were searched. 
```{r}
# Search genes in network
MA_NSL_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = MA_net.name, genes = NSL_ENSG, which.one = "new", net = NULL))
MA_NSL_mod <- tibble::rownames_to_column(MA_NSL_mod, "ENSG")
MA_NSL_mod <- dplyr::select(MA_NSL_mod, ENSG, module, p.val.mods)
MA_NSL_mod$Gene <- MA_NSL_mod$ENSG
# Add gene symbols
for (i in NSL_list$UKBEC_symbol) {
  loc <- match(i, NSL_list$UKBEC_symbol)
  MA_NSL_mod$Gene[MA_NSL_mod$Gene== i] <- NSL_list$Gene_symbol[loc]
}
# Get module membership
#MA_MM <- as.data.frame(MA_net$MM)
#colnames(MA_MM)<- "MM"
#MA_MM <- rownames_to_column(MA_MM, "ENSG")
#MA_MM$MM <- signif(MA_MM$MM, 4)
# Add MM to table
#MA_NSL_mod$MM <- MA_NSL_mod$ENSG
#for (i in MA_MM$ENSG) {
#  loc <- match(i, MA_MM$ENSG)
#  MA_NSL_mod$MM[MA_NSL_mod$MM== i] <- MA_MM$MM[loc]
#}
# Reorder table
MA_NSL_mod <- MA_NSL_mod[,c(4,1,2,3)]
MA_NSL_mod <- dplyr::arrange(MA_NSL_mod, p.val.mods)
# Store NSL enriched modules
MA_NSL_modules <- as.character(unique(MA_NSL_mod$module))
MA_NSL_modules <- MA_NSL_modules[!MA_NSL_modules =="void"]
# Print result
MA_NSL_mod
```
```{r}
# Remove void
MA_NSL_mod <- dplyr::filter(MA_NSL_mod, module !="void")
# Format dataframe for figure
MA_NSL_mod$FDR <- p.adjust(MA_NSL_mod$p.val.mods, method = "fdr")
names(MA_NSL_mod)[names(MA_NSL_mod)=="p.val.mods"] <- "p.val"
names(MA_NSL_mod)[names(MA_NSL_mod)=="module"] <- "Module"
MA_NSL_mod$p.val <- as.numeric(MA_NSL_mod$p.val)
MA_NSL_mod <- dplyr::arrange(MA_NSL_mod, Module)
MA_NSL_mod$`-log10(p.val)` <- -log10(MA_NSL_mod$p.val)
MA_NSL_mod$`-log10(FDR)` <- -log10(MA_NSL_mod$FDR)
# Produce plot
ggplot(MA_NSL_mod, aes(Gene, Module)) +
  geom_tile(aes(fill = `-log10(FDR)`)) +
  geom_text(aes(label = signif.num(FDR))) +
  scale_fill_gradient(low = "white", high = "goldenrod3", limits = c(0,4)) +
  labs(y = "Module", x = "Gene")
```

The number of genes in the output table was **`r nrow(MA_NSL_mod)`**, indicating there were **`r length(NSL_ENSG) - nrow(MA_NSL_mod)`** genes missing from the search. These genes appeared in `r length(unique(MA_NSL_modules))` modules, each of which was examined for **enrichment** of cell type specific markers. 
```{r, results = FALSE}
# Load cell type markers enriched for each module
MA_mod_enrich <- as.data.frame(getModulesEnrichment(MA_net, markers.path = file.path(here::here("data", "GMSCA_source", "markers"))))
# Filter enrichments for NSL modules
MA_mod_enrich_NSL <- dplyr::select(MA_mod_enrich, one_of(MA_NSL_modules))
```

The modules containing NSL genes showed **no significant enrichment** for **any cell types**.

### PD mendelian genes
The results tables for the mendelian PD gene list were filtered for the `r length(MA_NSL_modules)` modules containing NSL genes. 
```{r}
# Search genes in network
MA_PD_mendelian_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = MA_net.name, genes = PD_mendelian_ENSG, which.one = "new", net = NULL))
MA_PD_mendelian_mod <- tibble::rownames_to_column(MA_PD_mendelian_mod, "ENSG")
MA_PD_mendelian_mod <- dplyr::select(MA_PD_mendelian_mod, ENSG, module, p.val.mods)
MA_PD_mendelian_mod$Gene <- MA_PD_mendelian_mod$ENSG
# Add gene symbols
for (i in PD_mendelian_list$UKBEC_symbol) {
  loc <- match(i, PD_mendelian_list$UKBEC_symbol)
  MA_PD_mendelian_mod$Gene[MA_PD_mendelian_mod$Gene== i] <- PD_mendelian_list$Gene_symbol[loc]
}
# Add MM to table
#MA_PD_mendelian_mod$MM <- MA_PD_mendelian_mod$ENSG
#for (i in MA_MM$ENSG) {
#  loc <- match(i, MA_MM$ENSG)
#  MA_PD_mendelian_mod$MM[MA_PD_mendelian_mod$MM== i] <- MA_MM$MM[loc]
#}
# Reorder table
MA_PD_mendelian_mod <- MA_PD_mendelian_mod[,c(4,1,2,3)]
MA_PD_mendelian_mod <- dplyr::arrange(MA_PD_mendelian_mod, p.val.mods)
# Filter for NSL enriched modules 
MA_PD_mendelian_mod1 <- filter(MA_PD_mendelian_mod, module %in% MA_NSL_modules)
# Print result
MA_PD_mendelian_mod1
```

The number of genes in the output table was **`r nrow(MA_PD_mendelian_mod)`**, indicating there were **`r length(PD_mendelian_ENSG) - nrow(MA_PD_mendelian_mod)`** genes missing from the search.   

### PD sporadic genes
The results tables for the sporadic PD gene list were filtered for the `r length(MA_NSL_modules)` modules containing NSL genes. 
```{r}
# Search genes in network
MA_PD_GWAS_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = MA_net.name, genes = PD_GWAS_ENSG, which.one = "new", net = NULL))
MA_PD_GWAS_mod <- tibble::rownames_to_column(MA_PD_GWAS_mod, "ENSG")
MA_PD_GWAS_mod <- dplyr::select(MA_PD_GWAS_mod, ENSG, module, p.val.mods)
MA_PD_GWAS_mod$Gene <- MA_PD_GWAS_mod$ENSG
# Add gene symbols
for (i in PD_GWAS_list$UKBEC_symbol) {
  loc <- match(i, PD_GWAS_list$UKBEC_symbol)
  MA_PD_GWAS_mod$Gene[MA_PD_GWAS_mod$Gene== i] <- PD_GWAS_list$Gene_symbol[loc]
}
# Add MM to table
#MA_PD_GWAS_mod$MM <- MA_PD_GWAS_mod$ENSG
#for (i in MA_MM$ENSG) {
#  loc <- match(i, MA_MM$ENSG)
#  MA_PD_GWAS_mod$MM[MA_PD_GWAS_mod$MM== i] <- MA_MM$MM[loc]
#}
# Reorder table
MA_PD_GWAS_mod <- MA_PD_GWAS_mod[,c(4,1,2,3)]
MA_PD_GWAS_mod <- dplyr::arrange(MA_PD_GWAS_mod, p.val.mods)
# Filter for NSL enriched modules 
MA_PD_GWAS_mod1 <- filter(MA_PD_GWAS_mod, module %in% MA_NSL_modules)
# Print result
MA_PD_GWAS_mod1
```

The number of genes in the output table was **`r nrow(MA_PD_GWAS_mod)`**, indicating there were **`r length(PD_GWAS_ENSG) - nrow(MA_PD_GWAS_mod)`** genes missing from the search.  

## 4.2. Summary
The **frequency** of **genes** within each gene **list** apppearing in each of the **`r length(MA_NSL_modules)` NSL enriched modules** was calculated. 
```{r}
# Collate module results into one table
MA1 <- MA_PD_mendelian_mod1['module']
MA1['Gene_list'] <- "PD_mendelian"
MA2 <- MA_PD_GWAS_mod1['module']
MA2['Gene_list'] <- "PD_GWAS"
MA_mods_list <- do.call(rbind, list(MA1, MA2))
# Display as frequency of each module within each list
MA_mod_freq <- with(MA_mods_list, table(module, Gene_list))
MA_mod_freq <- as.data.frame.matrix(MA_mod_freq)
MA_mod_freq <- dplyr::filter(MA_mod_freq, rownames(MA_mod_freq) %in% MA_NSL_modules)
MA_mod_freq <- MA_mod_freq[c(1, 5, 2, 3, 4),]
```

```{r}
knitr::kable(MA_mod_freq)
```

The **significance** of the **enrichments** of each **gene list** within each of the **`r length(MA_NSL_modules)` NSL enriched modules** was also calculated. P-values were **FDR-corrected** before summarising. 
```{r}
# FDR correct p values
MA_PD_mendelian_mod1$FDR <- p.adjust(MA_PD_mendelian_mod1$p.val.mods, method = "fdr")
MA_PD_GWAS_mod1$FDR <- p.adjust(MA_PD_GWAS_mod1$p.val.mods, method = "fdr")
```

```{r, include = FALSE}
# Collate module results into one table
MAs1 <- data.frame(MA_PD_mendelian_mod1$module, MA_PD_mendelian_mod1$FDR)
names(MAs1)[names(MAs1)=="MA_PD_mendelian_mod1.FDR"] <- "FDR"
names(MAs1)[names(MAs1)=="MA_PD_mendelian_mod1.module"] <- "Module"
MAs1$Gene_list <- "PD_mendelian"
MAs2 <- data.frame(MA_PD_GWAS_mod1$module, MA_PD_GWAS_mod1$FDR)
names(MAs2)[names(MAs2)=="MA_PD_GWAS_mod1.FDR"] <- "FDR"
names(MAs2)[names(MAs2)=="MA_PD_GWAS_mod1.module"] <- "Module"
MAs2$Gene_list <- "PD_GWAS"
MA_pvals_list <- dplyr::bind_rows(MAs1, MAs2)
# Display as true/false table
MA_TF <- MA_pvals_list %>% 
  dplyr::distinct() %>%
  tidyr::spread(Gene_list, FDR)
MA_TF <- tibble::column_to_rownames(MA_TF, var = "Module")
MA_TF <- MA_TF[c(1, 5, 2, 3, 4),]
```

```{r}
knitr::kable(MA_TF)
```

```{r}
# Filter repeats
MA_melt_summ <- MA_pvals_list %>%
  dplyr::distinct()
# Prep for figure 
MA_melt_summ$Module <- as.character(MA_melt_summ$Module)
MA_melt_summ$Gene_list <- as.character(MA_melt_summ$Gene_list)
MA_melt_summ <- dplyr::arrange(MA_melt_summ, Module)
MA_melt_summ$`-log10(FDR)` <- -log10(MA_melt_summ$FDR)
# Plot
ggplot(MA_melt_summ, aes(Gene_list, Module)) +
  geom_tile(aes(fill = `-log10(FDR)`)) +
  geom_text(aes(label = signif.num(FDR))) +
  scale_fill_gradient(low = "white", high = "lightsteelblue4", limits = c(0,2)) +
  labs(y = "Module", x = "Gene list")
```

# 5. Microglia network
The overall structure of the network was first examined. 
```{r}
# Load network
MG_net = readRDS(file.path(here::here("data", "Networks", "netFCTX.12.signed.it.20.rds.No.MG.rds")))
# Plot sizes of modules making up the network
CoExpNets::plotModSizes(tissue = MG_net, which.one = "new")
```

```{r}
# Obtain eigengenes for each module
MG_net.name <- file.path(here::here("data", "Networks", "netFCTX.12.signed.it.20.rds.No.MG.rds"))
MG_egs <- CoExpNets::getNetworkEigengenes(tissue = MG_net.name, which.one = "new")
# Plot eigengene clustering 
CoExpNets::plotEGClustering(tissue = MG_net.name, which.one = "new")
```

```{r}
# Plot correlation between modules
corrplot(cor(MG_egs), tl.srt = 45,
         tl.col = "black",
         type = "upper", 
         tl.cex = 0.45,
         order = "hclust",
         title = paste0("Eigengene correlations"),
         mar = c(3, 2, 2.5, 4))
```

## 5.1. Searching gene lists {.tabset}
### NSL genes
```{r}
# Search genes in network
MG_NSL_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = MG_net.name, genes = NSL_ENSG, which.one = "new", net = NULL))
MG_NSL_mod <- tibble::rownames_to_column(MG_NSL_mod, "ENSG")
MG_NSL_mod <- dplyr::select(MG_NSL_mod, ENSG, module, p.val.mods)
MG_NSL_mod$Gene <- MG_NSL_mod$ENSG
# Add gene symbols
for (i in NSL_list$UKBEC_symbol) {
  loc <- match(i, NSL_list$UKBEC_symbol)
  MG_NSL_mod$Gene[MG_NSL_mod$Gene== i] <- NSL_list$Gene_symbol[loc]
}
# Get module membership
MG_MM <- as.data.frame(MG_net$MM)
colnames(MG_MM)<- "MM"
MG_MM <- rownames_to_column(MG_MM, "ENSG")
MG_MM$MM <- signif(MG_MM$MM, 4)
# Add MM to table
MG_NSL_mod$MM <- MG_NSL_mod$ENSG
for (i in MG_MM$ENSG) {
  loc <- match(i, MG_MM$ENSG)
  MG_NSL_mod$MM[MG_NSL_mod$MM== i] <- MG_MM$MM[loc]
}
# Reorder table
MG_NSL_mod <- MG_NSL_mod[,c(4,1,5,2,3)]
MG_NSL_mod <- dplyr::arrange(MG_NSL_mod, p.val.mods)
# Store NSL enriched modules
MG_NSL_modules <- as.character(unique(MG_NSL_mod$module))
MG_NSL_modules <- MG_NSL_modules[!MG_NSL_modules =="void"]
# Print result
MG_NSL_mod
```
```{r}
# Remove void
MG_NSL_mod <- dplyr::filter(MG_NSL_mod, module !="void")
# Format dataframe for figure
MG_NSL_mod$FDR <- p.adjust(MG_NSL_mod$p.val.mods, method = "fdr")
names(MG_NSL_mod)[names(MG_NSL_mod)=="p.val.mods"] <- "p.val"
names(MG_NSL_mod)[names(MG_NSL_mod)=="module"] <- "Module"
MG_NSL_mod$p.val <- as.numeric(MG_NSL_mod$p.val)
MG_NSL_mod <- dplyr::arrange(MG_NSL_mod, Module)
MG_NSL_mod$`-log10(p.val)` <- -log10(MG_NSL_mod$p.val)
MG_NSL_mod$`-log10(FDR)` <- -log10(MG_NSL_mod$FDR)
# Produce plot
ggplot(MG_NSL_mod, aes(Gene, Module)) +
  geom_tile(aes(fill = `-log10(FDR)`)) +
  geom_text(aes(label = signif.num(FDR))) +
  scale_fill_gradient(low = "white", high = "goldenrod3", limits = c(0,4)) +
  labs(y = "Module", x = "Gene")
```

The number of genes in the output table was **`r nrow(MG_NSL_mod)`**, indicating there were **`r length(NSL_ENSG) - nrow(MG_NSL_mod)`** genes missing from the search. These genes appeared in `r length(unique(MG_NSL_modules))` modules, each of which was examined for **enrichment** of cell type specific markers. 
```{r, results = FALSE}
# Load cell type markers enriched for each module
MG_mod_enrich <- as.data.frame(getModulesEnrichment(MG_net, markers.path = file.path(here::here("data", "GMSCA_source", "markers"))))
# Filter enrichments for NSL modules
MG_mod_enrich_NSL <- dplyr::select(MG_mod_enrich, one_of(MG_NSL_modules))
```

The modules containing NSL genes showed **no significant enrichment** for **any cell types**. 

### PD mendelian genes
The results tables for the mendelian PD gene list were filtered for the `r length(MG_NSL_modules)` modules containing NSL genes. 
```{r}
# Search genes in network
MG_PD_mendelian_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = MG_net.name, genes = PD_mendelian_ENSG, which.one = "new", net = NULL))
MG_PD_mendelian_mod <- tibble::rownames_to_column(MG_PD_mendelian_mod, "ENSG")
MG_PD_mendelian_mod <- dplyr::select(MG_PD_mendelian_mod, ENSG, module, p.val.mods)
MG_PD_mendelian_mod$Gene <- MG_PD_mendelian_mod$ENSG
# Add gene symbols
for (i in PD_mendelian_list$UKBEC_symbol) {
  loc <- match(i, PD_mendelian_list$UKBEC_symbol)
  MG_PD_mendelian_mod$Gene[MG_PD_mendelian_mod$Gene== i] <- PD_mendelian_list$Gene_symbol[loc]
}
# Add MM to table
MG_PD_mendelian_mod$MM <- MG_PD_mendelian_mod$ENSG
for (i in MG_MM$ENSG) {
  loc <- match(i, MG_MM$ENSG)
  MG_PD_mendelian_mod$MM[MG_PD_mendelian_mod$MM== i] <- MG_MM$MM[loc]
}
# Reorder table
MG_PD_mendelian_mod <- MG_PD_mendelian_mod[,c(4,1,5,2,3)]
MG_PD_mendelian_mod <- dplyr::arrange(MG_PD_mendelian_mod, p.val.mods)
# Filter for NSL enriched modules 
MG_PD_mendelian_mod1 <- filter(MG_PD_mendelian_mod, module %in% MG_NSL_modules)
# Print result
MG_PD_mendelian_mod1
```

The number of genes in the output table was **`r nrow(MG_PD_mendelian_mod)`**, indicating there were **`r length(PD_mendelian_ENSG) - nrow(MG_PD_mendelian_mod)`** genes missing from the search.   

### PD sporadic genes
The results tables for the sporadic PD gene list were filtered for the `r length(MG_NSL_modules)` modules containing NSL genes. 
```{r}
# Search genes in network
MG_PD_GWAS_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = MG_net.name, genes = PD_GWAS_ENSG, which.one = "new", net = NULL))
MG_PD_GWAS_mod <- tibble::rownames_to_column(MG_PD_GWAS_mod, "ENSG")
MG_PD_GWAS_mod <- dplyr::select(MG_PD_GWAS_mod, ENSG, module, p.val.mods)
MG_PD_GWAS_mod$Gene <- MG_PD_GWAS_mod$ENSG
# Add gene symbols
for (i in PD_GWAS_list$UKBEC_symbol) {
  loc <- match(i, PD_GWAS_list$UKBEC_symbol)
  MG_PD_GWAS_mod$Gene[MG_PD_GWAS_mod$Gene== i] <- PD_GWAS_list$Gene_symbol[loc]
}
# Add MM to table
MG_PD_GWAS_mod$MM <- MG_PD_GWAS_mod$ENSG
for (i in MG_MM$ENSG) {
  loc <- match(i, MG_MM$ENSG)
  MG_PD_GWAS_mod$MM[MG_PD_GWAS_mod$MM== i] <- MG_MM$MM[loc]
}
# Reorder table
MG_PD_GWAS_mod <- MG_PD_GWAS_mod[,c(4,1,5,2,3)]
MG_PD_GWAS_mod <- dplyr::arrange(MG_PD_GWAS_mod, p.val.mods)
# Filter for NSL enriched modules 
MG_PD_GWAS_mod1 <- filter(MG_PD_GWAS_mod, module %in% MG_NSL_modules)
# Print result
MG_PD_GWAS_mod1
```

The number of genes in the output table was **`r nrow(MG_PD_GWAS_mod)`**, indicating there were **`r length(PD_GWAS_ENSG) - nrow(MG_PD_GWAS_mod)`** genes missing from the search.  

## 5.2. Summary
The **frequency** of **genes** within each gene **list** apppearing in each of the  **`r length(MG_NSL_modules)` NSL enriched modules** was calculated. 
```{r, include = FALSE}
# Collate module results into one table
MG1 <- MG_PD_mendelian_mod1['module']
MG1['Gene_list'] <- "PD_mendelian"
MG2 <- MG_PD_GWAS_mod1['module']
MG2['Gene_list'] <- "PD_GWAS"
MG_mods_list <- do.call(rbind, list(MG1, MG2))
# Display as frequency of each module within each list
MG_mod_freq <- with(MG_mods_list, table(module, Gene_list))
MG_mod_freq <- as.data.frame.matrix(MG_mod_freq)
MG_mod_freq <- dplyr::filter(MG_mod_freq, rownames(MG_mod_freq) %in% MG_NSL_modules)
MG_mod_freq <- MG_mod_freq[c(5, 3, 4, 2, 1),]
```

```{r}
knitr::kable(MG_mod_freq)
```

The **significance** of the **enrichments** of each **gene list** within each of the **`r length(MG_NSL_modules)` NSL enriched modules** was also calculated. P-values were **FDR-corrected** before summarising. 
```{r}
# FDR correct p values
MG_PD_mendelian_mod1$FDR <- p.adjust(MG_PD_mendelian_mod1$p.val.mods, method = "fdr")
MG_PD_GWAS_mod1$FDR <- p.adjust(MG_PD_GWAS_mod1$p.val.mods, method = "fdr")
```

```{r, include = FALSE}
# Collate module results into one table
MGs1 <- data.frame(MG_PD_mendelian_mod1$module, MG_PD_mendelian_mod1$FDR)
names(MGs1)[names(MGs1)=="MG_PD_mendelian_mod1.FDR"] <- "FDR"
names(MGs1)[names(MGs1)=="MG_PD_mendelian_mod1.module"] <- "Module"
MGs1$Gene_list <- "PD_mendelian"
MGs2 <- data.frame(MG_PD_GWAS_mod1$module, MG_PD_GWAS_mod1$FDR)
names(MGs2)[names(MGs2)=="MG_PD_GWAS_mod1.FDR"] <- "FDR"
names(MGs2)[names(MGs2)=="MG_PD_GWAS_mod1.module"] <- "Module"
MGs2$Gene_list <- "PD_GWAS"
MG_pvals_list <- dplyr::bind_rows(MGs1, MGs2)
# Display as true/false table
MG_TF <- MG_pvals_list %>% 
  dplyr::distinct() %>%
  tidyr::spread(Gene_list, FDR)
MG_TF <- tibble::column_to_rownames(MG_TF, var = "Module")
MG_TF <- MG_TF[c(5, 3, 4, 2, 1), ]
```

```{r}
knitr::kable(MG_TF)
```

```{r}
# Filter repeats
MG_melt_summ <- MG_pvals_list %>%
  dplyr::distinct()
# Prep for figure 
MG_melt_summ$Module <- as.character(MG_melt_summ$Module)
MG_melt_summ$Gene_list <- as.character(MG_melt_summ$Gene_list)
MG_melt_summ <- dplyr::arrange(MG_melt_summ, Module)
MG_melt_summ$`-log10(FDR)` <- -log10(MG_melt_summ$FDR)
# Plot
ggplot(MG_melt_summ, aes(Gene_list, Module)) +
  geom_tile(aes(fill = `-log10(FDR)`)) +
  geom_text(aes(label = signif.num(FDR))) +
  scale_fill_gradient(low = "white", high = "lightsteelblue4", limits = c(0,2)) +
  labs(y = "Module", x = "Gene list")
```

# 6. Oligodendrocyte network
The overall structure of the network was first examined.
```{r}
# Load network
OLG_net = readRDS(file.path(here::here("data", "Networks", "netFCTX.12.signed.it.20.rds.No.OLG.rds")))
# Plot sizes of modules making up the network
CoExpNets::plotModSizes(tissue = OLG_net, which.one = "new")
```

```{r}
# Obtain eigengenes for each module
OLG_net.name <- file.path(here::here("data", "Networks", "netFCTX.12.signed.it.20.rds.No.OLG.rds"))
OLG_egs <- CoExpNets::getNetworkEigengenes(tissue = OLG_net.name, which.one = "new")
# Plot eigengene clustering 
CoExpNets::plotEGClustering(tissue = OLG_net.name, which.one = "new")
```

```{r}
# Plot correlation between modules
corrplot(cor(OLG_egs), tl.srt = 45,
         tl.col = "black",
         type = "upper", 
         tl.cex = 0.45,
         order = "hclust",
         title = paste0("Eigengene correlations for UKBEC modules"),
         mar = c(3, 2, 2.5, 4))
```

## 6.1. Searching gene lists {.tabset}
### NSL genes
```{r}
# Search genes in network
OLG_NSL_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = OLG_net.name, genes = NSL_ENSG, which.one = "new", net = NULL))
OLG_NSL_mod <- tibble::rownames_to_column(OLG_NSL_mod, "ENSG")
OLG_NSL_mod <- dplyr::select(OLG_NSL_mod, ENSG, module, p.val.mods)
OLG_NSL_mod$Gene <- OLG_NSL_mod$ENSG
# Add gene symbols
for (i in NSL_list$UKBEC_symbol) {
  loc <- match(i, NSL_list$UKBEC_symbol)
  OLG_NSL_mod$Gene[OLG_NSL_mod$Gene== i] <- NSL_list$Gene_symbol[loc]
}
# Get module membership
OLG_MM <- as.data.frame(OLG_net$MM)
colnames(OLG_MM)<- "MM"
OLG_MM <- rownames_to_column(OLG_MM, "ENSG")
OLG_MM$MM <- signif(OLG_MM$MM, 4)
# Add MM to table
OLG_NSL_mod$MM <- OLG_NSL_mod$ENSG
for (i in OLG_MM$ENSG) {
  loc <- match(i, OLG_MM$ENSG)
  OLG_NSL_mod$MM[OLG_NSL_mod$MM== i] <- OLG_MM$MM[loc]
}
# Reorder table
OLG_NSL_mod <- OLG_NSL_mod[,c(4,1,5,2,3)]
OLG_NSL_mod <- dplyr::arrange(OLG_NSL_mod, p.val.mods)
# Store NSL enriched modules
OLG_NSL_modules <- as.character(unique(OLG_NSL_mod$module))
OLG_NSL_modules <- OLG_NSL_modules[!OLG_NSL_modules =="void"]
# Print result
OLG_NSL_mod
```
```{r}
# Remove void
OLG_NSL_mod <- dplyr::filter(OLG_NSL_mod, module !="void")
# Format dataframe for figure
OLG_NSL_mod$FDR <- p.adjust(OLG_NSL_mod$p.val.mods, method = "fdr")
names(OLG_NSL_mod)[names(OLG_NSL_mod)=="p.val.mods"] <- "p.val"
names(OLG_NSL_mod)[names(OLG_NSL_mod)=="module"] <- "Module"
OLG_NSL_mod$p.val <- as.numeric(OLG_NSL_mod$p.val)
OLG_NSL_mod <- dplyr::arrange(OLG_NSL_mod, Module)
OLG_NSL_mod$`-log10(p.val)` <- -log10(OLG_NSL_mod$p.val)
OLG_NSL_mod$`-log10(FDR)` <- -log10(OLG_NSL_mod$FDR)
# Produce plot
ggplot(OLG_NSL_mod, aes(Gene, Module)) +
  geom_tile(aes(fill = `-log10(FDR)`)) +
  geom_text(aes(label = signif.num(FDR))) +
  scale_fill_gradient(low = "white", high = "goldenrod3", limits = c(0,4)) +
  labs(y = "Module", x = "Gene")
```

The number of genes in the output table was **`r nrow(OLG_NSL_mod)`**, indicating there were **`r length(NSL_ENSG) - nrow(OLG_NSL_mod)`** genes missing from the search. These genes appeared in `r length(unique(OLG_NSL_modules))` modules, each of which was examined for **enrichment** of cell type specific markers. 
```{r, results = FALSE}
# Load cell type markers enriched for each module
OLG_mod_enrich <- as.data.frame(getModulesEnrichment(OLG_net, markers.path = file.path(here::here("data", "GMSCA_source", "markers"))))
# Filter enrichments for NSL modules
OLG_mod_enrich_NSL <- dplyr::select(OLG_mod_enrich, one_of(OLG_NSL_modules))
```

The modules containing NSL genes showed **one significant enrichment** for a **neuronal cell type**.

### PD mendelian genes
```{r}
# Search genes in network
OLG_PD_mendelian_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = OLG_net.name, genes = PD_mendelian_ENSG, which.one = "new", net = NULL))
OLG_PD_mendelian_mod <- tibble::rownames_to_column(OLG_PD_mendelian_mod, "ENSG")
OLG_PD_mendelian_mod <- dplyr::select(OLG_PD_mendelian_mod, ENSG, module, p.val.mods)
OLG_PD_mendelian_mod$Gene <- OLG_PD_mendelian_mod$ENSG
# Add gene symbols
for (i in PD_mendelian_list$UKBEC_symbol) {
  loc <- match(i, PD_mendelian_list$UKBEC_symbol)
  OLG_PD_mendelian_mod$Gene[OLG_PD_mendelian_mod$Gene== i] <- PD_mendelian_list$Gene_symbol[loc]
}
# Add MM to table
OLG_PD_mendelian_mod$MM <- OLG_PD_mendelian_mod$ENSG
for (i in OLG_MM$ENSG) {
  loc <- match(i, OLG_MM$ENSG)
  OLG_PD_mendelian_mod$MM[OLG_PD_mendelian_mod$MM== i] <- OLG_MM$MM[loc]
}
# Reorder table
OLG_PD_mendelian_mod <- OLG_PD_mendelian_mod[,c(4,1,5,2,3)]
OLG_PD_mendelian_mod <- dplyr::arrange(OLG_PD_mendelian_mod, p.val.mods)
# Filter for NSL enriched modules 
OLG_PD_mendelian_mod1 <- filter(OLG_PD_mendelian_mod, module %in% OLG_NSL_modules)
# Print result
OLG_PD_mendelian_mod1
```
The number of genes in the output table was **`r nrow(OLG_PD_mendelian_mod)`**, indicating there were **`r length(PD_mendelian_ENSG) - nrow(OLG_PD_mendelian_mod)`** genes missing from the search.  

### PD sporadic genes
```{r}
# Search genes in network
OLG_PD_GWAS_mod <- as.data.frame(CoExpNets::reportOnFETGenes(tissue = OLG_net.name, genes = PD_GWAS_ENSG, which.one = "new", net = NULL))
OLG_PD_GWAS_mod <- tibble::rownames_to_column(OLG_PD_GWAS_mod, "ENSG")
OLG_PD_GWAS_mod <- dplyr::select(OLG_PD_GWAS_mod, ENSG, module, p.val.mods)
OLG_PD_GWAS_mod$Gene <- OLG_PD_GWAS_mod$ENSG
# Add gene symbols
for (i in PD_GWAS_list$UKBEC_symbol) {
  loc <- match(i, PD_GWAS_list$UKBEC_symbol)
  OLG_PD_GWAS_mod$Gene[OLG_PD_GWAS_mod$Gene== i] <- PD_GWAS_list$Gene_symbol[loc]
}
# Add MM to table
OLG_PD_GWAS_mod$MM <- OLG_PD_GWAS_mod$ENSG
for (i in OLG_MM$ENSG) {
  loc <- match(i, OLG_MM$ENSG)
  OLG_PD_GWAS_mod$MM[OLG_PD_GWAS_mod$MM== i] <- OLG_MM$MM[loc]
}
# Reorder table
OLG_PD_GWAS_mod <- OLG_PD_GWAS_mod[,c(4,1,5,2,3)]
OLG_PD_GWAS_mod <- dplyr::arrange(OLG_PD_GWAS_mod, p.val.mods)
# Filter for NSL enriched modules 
OLG_PD_GWAS_mod1 <- filter(OLG_PD_GWAS_mod, module %in% OLG_NSL_modules)
# Print result
OLG_PD_GWAS_mod1
```
The number of genes in the output table was **`r nrow(OLG_PD_GWAS_mod)`**, indicating there were **`r length(PD_GWAS_ENSG) - nrow(OLG_PD_GWAS_mod)`** genes missing from the search.  

## 6.2. Summary
The **frequency** of **genes** within each gene **list** apppearing in each of the **`r length(OLG_NSL_modules)` NSL enriched modules** was calculated. 
```{r}
# Collate module results into one table
OLG1 <- OLG_PD_mendelian_mod1['module']
OLG1['Gene_list'] <- "PD_mendelian"
OLG2 <- OLG_PD_GWAS_mod1['module']
OLG2['Gene_list'] <- "PD_GWAS"
OLG_mods_list <- do.call(rbind, list(OLG1, OLG2))
# Display as frequency of each module within each list
OLG_mod_freq <- with(OLG_mods_list, table(module, Gene_list))
OLG_mod_freq <- as.data.frame.matrix(OLG_mod_freq)
OLG_mod_freq <- dplyr::filter(OLG_mod_freq, rownames(OLG_mod_freq) %in% OLG_NSL_modules)
OLG_mod_freq <- OLG_mod_freq[c(2, 1, 4, 5, 3),]
```

```{r}
knitr::kable(OLG_mod_freq)
```

The **significance** of the **enrichments** of each **gene list** within each of the **`r length(OLG_NSL_modules)` NSL enriched modules** was also calculated. P-values were **FDR-corrected** before summarising. 
```{r}
# FDR correct p values
OLG_PD_mendelian_mod1$FDR <- p.adjust(OLG_PD_mendelian_mod1$p.val.mods, method = "fdr")
OLG_PD_GWAS_mod1$FDR <- p.adjust(OLG_PD_GWAS_mod1$p.val.mods, method = "fdr")
```

```{r}
# Collate module results into one table
OLGs1 <- data.frame(OLG_PD_mendelian_mod1$module, OLG_PD_mendelian_mod1$FDR)
names(OLGs1)[names(OLGs1)=="OLG_PD_mendelian_mod1.FDR"] <- "FDR"
names(OLGs1)[names(OLGs1)=="OLG_PD_mendelian_mod1.module"] <- "Module"
OLGs1$Gene_list <- "PD_mendelian"
OLGs2 <- data.frame(OLG_PD_GWAS_mod1$module, OLG_PD_GWAS_mod1$FDR)
names(OLGs2)[names(OLGs2)=="OLG_PD_GWAS_mod1.FDR"] <- "FDR"
names(OLGs2)[names(OLGs2)=="OLG_PD_GWAS_mod1.module"] <- "Module"
OLGs2$Gene_list <- "PD_GWAS"
OLG_pvals_list <- dplyr::bind_rows(OLGs1, OLGs2)
# Display as true/false table
OLG_TF <- OLG_pvals_list %>% 
  dplyr::distinct() %>%
  tidyr::spread(Gene_list, FDR)
OLG_TF <- tibble::column_to_rownames(OLG_TF, var = "Module")
OLG_TF <- OLG_TF[c(2, 1, 5, 3, 4), ]
```

```{r}
knitr::kable(OLG_TF)
```

```{r}
# Filter repeats
OLG_melt_summ <- OLG_pvals_list %>%
  dplyr::distinct()
# Prep for figure 
OLG_melt_summ$Module <- as.character(OLG_melt_summ$Module)
OLG_melt_summ$Gene_list <- as.character(OLG_melt_summ$Gene_list)
OLG_melt_summ <- dplyr::arrange(OLG_melt_summ, Module)
OLG_melt_summ$`-log10(FDR)` <- -log10(OLG_melt_summ$FDR)
# Plot
ggplot(OLG_melt_summ, aes(Gene_list, Module)) +
  geom_tile(aes(fill = `-log10(FDR)`)) +
  geom_text(aes(label = signif.num(FDR))) +
  scale_fill_gradient(low = "white", high = "lightsteelblue4") +
  labs(y = "Module", x = "Gene list")
```

# 7. Overall summary
First, the overall clustering of the NSL genes was examined
```{r}
N_NSL_mod$Network <- "Neuron-corrected"
MA_NSL_mod$Network <- "Astrocyte-corrected"
MG_NSL_mod$Network <- "Microglia-corrected"
OLG_NSL_mod$Network <- "Oligodendrocyte-corrected"
NSL_fig <- do.call(rbind, list(N_NSL_mod[,c(1,2,4,5,6,7,8,9)],
                               MA_NSL_mod, 
                               MG_NSL_mod[,c(1,2,4,5,6,7,8,9)],
                               OLG_NSL_mod[,c(1,2,4,5,6,7,8,9)]))
NSL_fig <- dplyr::select(NSL_fig, Gene, Module, FDR, `-log10(FDR)`, Network)
# Produce plot
ggplot(NSL_fig, aes(Gene, Module)) +
  geom_tile(aes(fill = `-log10(FDR)`)) +
  geom_text(aes(label = signif.num(FDR)), size = 2) +
  scale_fill_gradient(low = "white", high = "goldenrod3") +
  labs(y = "Module") +
  facet_grid(factor(Network, 
                    levels = c("Neuron-corrected", "Astrocyte-corrected", "Microglia-corrected", "Oligodendrocyte-corrected")) ~ ., 
             scales = "free", space = "free") +
  theme(strip.text.y = element_text(angle = 90, size = 6), axis.text = element_text(size = 6))
```

Each of the cell type corrections succeeded in increasing the number of cell type enrichments associated with GCN modules and the genes therein. These results were thus collated together to evaluate the cell type specific activity of the **NSL genes**.
```{r}
# Format cell type enrichment tables
N_mod_enrich_fig <- dplyr::filter_all(N_mod_enrich_NSL, any_vars(.<0.05))
N_mod_enrich_fig <- as.data.frame(t(N_mod_enrich_fig))
N_mod_enrich_fig <- rownames_to_column(N_mod_enrich_fig, var = "Module")
MA_mod_enrich_fig <- dplyr::filter_all(MA_mod_enrich_NSL, any_vars(.<0.05))
MA_mod_enrich_fig <- as.data.frame(t(MA_mod_enrich_fig))
MA_mod_enrich_fig <- rownames_to_column(MA_mod_enrich_fig, var = "Module")
MG_mod_enrich_fig <- dplyr::filter_all(MG_mod_enrich_NSL, any_vars(.<0.05))
MG_mod_enrich_fig <- as.data.frame(t(MG_mod_enrich_fig))
MG_mod_enrich_fig <- rownames_to_column(MG_mod_enrich_fig, var = "Module")
OLG_mod_enrich_fig <- dplyr::filter_all(OLG_mod_enrich_NSL, any_vars(.<0.05))
OLG_mod_enrich_fig <- as.data.frame(t(OLG_mod_enrich_fig))
OLG_mod_enrich_fig <- rownames_to_column(OLG_mod_enrich_fig, var = "Module")
ukbec_enrich_fig <- dplyr::filter_all(ukbec_enrich_NSL, any_vars(.<0.05)) 
ukbec_enrich_fig <- as.data.frame(t(ukbec_enrich_fig))
ukbec_enrich_fig <- rownames_to_column(ukbec_enrich_fig, var = "Module")
# Format NSL gene enrichment tables
N_NSL_fig <- dplyr::select(N_NSL_mod, Gene, Module)
N_NSL_fig$Network <- "Neuron-corrected"
MA_NSL_fig <- dplyr::select(MA_NSL_mod, Gene, Module)
MA_NSL_fig$Network <- "Astrocyte-corrected"
MG_NSL_fig <- dplyr::select(MG_NSL_mod, Gene, Module)
MG_NSL_fig$Network <- "Microglia-corrected"
OLG_NSL_fig <- dplyr::select(OLG_NSL_mod, Gene, Module)
OLG_NSL_fig$Network <- "Oligodendrocyte-corrected"
ukbec_NSL_fig <- dplyr::select(NSL_rep_uk, gene, module)
names(ukbec_NSL_fig)[names(ukbec_NSL_fig)=="module"] <- "Module"
names(ukbec_NSL_fig)[names(ukbec_NSL_fig)== "gene"] <- "Gene"
ukbec_NSL_fig$Network <- "None"
# Only one enrichment so can't join together
OLG_NSL_figure <- dplyr::full_join(OLG_mod_enrich_fig, OLG_NSL_fig, by = "Module")
OLG_NSL_figure <- dplyr::filter_all(OLG_NSL_figure, any_vars(.<1))
# Reformat
OLG_NSL_figure$`-log10(p.value)` <- -log10(OLG_NSL_figure$Neuron_Neuron.Ex1)
OLG_NSL_figure$`-log10(p.value)` <- as.numeric(OLG_NSL_figure$`-log10(p.value)`)
names(OLG_NSL_figure)[names(OLG_NSL_figure)=="Neuron_Neuron.Ex1"] <- "Enrichment"
OLG_NSL_figure$Cell_type <- "Neuron_Neuron.Ex1"
# Produce plot
ggplot(OLG_NSL_figure, aes(Gene, Cell_type)) +
  geom_tile(aes(fill = `-log10(p.value)`)) +
  geom_text(aes(label = signif.num2(Enrichment)), size = 3) +
  scale_fill_gradient(low = "white", high = "goldenrod3") +
  labs(y = "Cell type") +
  theme(axis.text = element_text(size = 6))
```