---
title: "LDSC"
author: "Amy Hicks"
date: "05/08/2022"
output:
  html_document:
    code_folding: hide
    toc: true
    toc_float: true
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(corrplot)
library(devtools)
library(DT)
library(here)
library(ggplot2)
library(GeneOverlap)
library(LDSCforRyten)
library(tidyverse)
library(readxl)
library(UpSetR)
library(CoExpNets)
library(doParallel)
library(foreach)
library(qdapTools)
library(stringr)
library(janitor)
library(readxl)
```

This markdown used linkage disequelibrium score regression (**LDSC**) to test whether the NSL regulons derived from **ARACNe-AP** analysis of GTEx data were enriched for **Parkinson's disease heritability**. The most recent [PD GWAS data](https://pubmed.ncbi.nlm.nih.gov/31701892/) was used, and LDSC was run on the ARACNe-AP results split up in two different ways: 
1. gene lists split up according to individual regulons (i.e. labelled HCFC1, KANSL1, KAT8, OGT, WDR5, MCRS1)
2. gene lists split up according to regulon cummulative frequency (i.e. labelled 1 or more, 2 or more, 3 or more, 4 or more, 5 or more, 6)

# 1. Preparing input files
As the output of ARACNe-AP was in the form of **ensembl IDs**, these were first translated into **gene symbols** and saved. 
```{r, eval=FALSE}
# Biomart needed to convert to gene names
ensembl = useMart(biomart = "ENSEMBL_MART_ENSEMBL", dataset = "hsapiens_gene_ensembl")
# Upload ARACNe results from GTEx dataset
aracne <- read.delim(file.path(here::here("data", "ARACNe", "gtex_expr.txt.aracnet.txt")))
aracne_filtered <- aracne %>%
  dplyr::select(Target) %>%
  dplyr::distinct()
# Get gene symbols for ENSGs
aracne_BMlist <- getBM(attributes = c("hgnc_symbol", "ensembl_gene_id"), 
                       filters = "ensembl_gene_id", 
                       values = aracne_filtered, 
                       mart = ensembl)
aracne_list <- aracne_BMlist %>%
  dplyr::distinct(hgnc_symbol, .keep_all = TRUE) %>%
  dplyr::filter(hgnc_symbol != "")
# See which are missing
aracne_missing <- dplyr::setdiff(aracne_filtered$Target, aracne_list$ensembl_gene_id)
# Search missing 
aracne_BMlist2 <- getBM(attributes = c("hgnc_symbol", "ensembl_gene_id"), 
                        filters = "ensembl_gene_id", 
                        values = aracne_missing, 
                        mart = ensembl)
aracne_list2 <- aracne_BMlist2 %>%
  dplyr::distinct(hgnc_symbol, .keep_all = TRUE) %>%
  dplyr::filter(hgnc_symbol != "")
# No gene names found, try lapply function (locates within GCN)
aracne_list3 <- as.data.frame(aracne_missing)
names(aracne_list3)[names(aracne_list3)=="aracne_missing"] <- "ensembl_gene_id"
aracne_list3$hgnc_symbol <- lapply(aracne_list3$ensembl_gene_id, CoExpNets::fromEnsembl2GeneName)
aracne_list3$hgnc_symbol <- as.character(aracne_list3$hgnc_symbol)
# Join onto big list 
aracne_fulllist <- dplyr::bind_rows(aracne_list, aracne_list3)
# Join on MI values
names(aracne)[names(aracne)=="Target"] <- "ensembl_gene_id"
aracne_fulllist <- dplyr::left_join(aracne, aracne_fulllist, by = "ensembl_gene_id")
names(aracne_fulllist)[names(aracne_fulllist)=="ensembl_gene_id"] <- "Target_ENSG"
names(aracne_fulllist)[names(aracne_fulllist)=="hgnc_symbol"] <- "Target_symbol"
names(aracne_fulllist)[names(aracne_fulllist)=="Regulator"] <- "Regulator_ENSG"
aracne_fulllist$Regulator_symbol <- as.character(aracne_fulllist$Regulator_ENSG)
for (i in NSL_list$Ensembl_ID) {
  loc <- match(i, NSL_list$Ensembl_ID)
  aracne_fulllist$Regulator_symbol[aracne_fulllist$Regulator_symbol == i] <- NSL_list$Gene_symbol[loc]
}
# Save table
write.csv(aracne_fulllist, file.path(here::here("data", "ARACNe", "aracne_list.csv")), row.names = FALSE)
```

```{r}
# Load table
aracne_fulllist <- read.csv(file.path(here::here("data", "ARACNe", "aracne_list.csv")))
```

The total number of regulator-regulon **pairs** (relationships) within this dataset is: **`r length(aracne_fulllist$MI)`**. The total number of **unique** genes predicted to be regulated by any of the 6 NSL genes is: **`r length(unique(aracne_fulllist$Target_ENSG))`**. This number is smaller as many of the genes appear in multiple regulons. 

# 2. Creating LDSC annotations
## 2.1. Obtaining gene position details
Next, the gene position details were obtained as well as the 100kb window either side. 
```{r, eval = FALSE}
# Use the ENSGs to combat missing genes
aracne_ldsc <- aracne_fulllist %>% 
  LDSCforRyten::AddGenePosDetails_RemoveXandYandMT(.,
                                                   columnToFilter = "Target_ENSG", 
                                                   mart = 37, 
                                                   attributes = c("ensembl_gene_id","chromosome_name", 
                                                                  "start_position","end_position"),
                                                   filter = c("ensembl_gene_id")) %>% 
  LDSCforRyten::AddBPWindow(windowsize = 100000)
# Save file
write.csv(aracne_ldsc, file.path(here::here("data", "ARACNe", "aracne_ldsc.csv")), row.names = FALSE)
```

```{r}
# Upload
aracne_ldsc <- read.csv(file.path(here::here("data", "ARACNe", "aracne_ldsc.csv")))
```

How many are **missing** from a total list of `r length(unique(aracne_fulllist$Target_ENSG))`?
```{r}
# How many are missing?
length(dplyr::setdiff(aracne_fulllist$Target_ENSG, aracne_ldsc$Target_ENSG))
```

```{r, eval = FALSE}
# Store missing
aracne_ldsc_missing <- dplyr::setdiff(aracne_fulllist$Target_ENSG, aracne_ldsc$Target_ENSG)
am <- getBM(attributes = c("ensembl_gene_id",
                           "chromosome_name",
                           "start_position",
                           "end_position"),       
            filter = c("ensembl_gene_id"), 
            values = aracne_ldsc_missing,
            mart = ensembl)
```

The remaining missing genes are located on **chromosome X**, which is not measured in GWAS and therefore cannot be studied using LDSC. 
```{r, eval = FALSE}
# Load baseline and creat G ranges object from baseline model
BM <- LDSCforRyten::creating_baseline_df(baseline_model = "97")
BM_GR <- LDSCforRyten::df2GRanges(BM, 
                                  seqname_col = "CHR", 
                                  start_col = "BP",
                                  end_col = "BP")
```

## 2.2. Splitting gene lists {.tabset}
### Individual regulons
Splitting the results according to regulon meant that there were overlaps between the lists, as some genes appeared in more than one regulon. 
```{r}
# How many genes are in each list?
aracne_ldsc %>%
  dplyr::count(Regulator_symbol)
```

```{r, eval = FALSE}
# Split results by individual regulons 
aracne_ldsc_list1 <- base::split(aracne_ldsc, f = aracne_ldsc$Regulator_symbol)
```

### Cummulative frequency
Splitting the results according to regulon cummulative frequency also gave rise to overlaps between the lists, as "1 or more" contained all genes. 
```{r}
# Split list by frequency of appearance in regulons
aracne_count <- aracne_ldsc %>%
  dplyr::count(Target_symbol)
aracne_ldsc_freq <- dplyr::left_join(aracne_ldsc, aracne_count, by = "Target_symbol")
names(aracne_ldsc_freq)[names(aracne_ldsc_freq)=="n"] <- "frequency"
# Filter for genes appearing in different numbers of regulons
aracne_ldsc_cm2 <- aracne_ldsc_freq %>%
  dplyr::filter(frequency > 1)
aracne_ldsc_cm3 <- aracne_ldsc_freq %>%
  dplyr::filter(frequency > 2)
aracne_ldsc_cm4 <- aracne_ldsc_freq %>%
  dplyr::filter(frequency > 3)
aracne_ldsc_cm5 <- aracne_ldsc_freq %>%
  dplyr::filter(frequency > 4)
aracne_ldsc_cm6 <- aracne_ldsc_freq %>%
  dplyr::filter(frequency > 5)
# List
aracne_ldsc_list2 <- list(aracne_ldsc_freq, 
                          aracne_ldsc_cm2, 
                          aracne_ldsc_cm3, 
                          aracne_ldsc_cm4, 
                          aracne_ldsc_cm5, 
                          aracne_ldsc_cm6)
names(aracne_ldsc_list2) <- c("min_1", "min_2", "min_3", "min_4", "min_5", "min_6")
# How many genes in each list?
nrow(aracne_ldsc_list2$min_1)
nrow(aracne_ldsc_list2$min_2)
nrow(aracne_ldsc_list2$min_3)
nrow(aracne_ldsc_list2$min_4)
nrow(aracne_ldsc_list2$min_5)
nrow(aracne_ldsc_list2$min_6)
```

The number of unique genes appearing in at least 3 regulons is: **`r length(unique(aracne_ldsc_cm3$Target_ENSG))`**. This amounts to `r length(unique(aracne_ldsc_cm3$Target_ENSG))*100/length(unique(aracne_fulllist$Target_ENSG))`% of the total list. 

## 2.3. Annotate lists
```{r, eval = FALSE}
# Find overlapping regions between genes in each annotation and BM
hits1 <- aracne_ldsc_list1 %>%
  LDSCforRyten::overlap_annot_list(BM_GR, 
                                   seqname_col = "chromosome_name", 
                                   start_col = "Gene.Start.MinusKB", 
                                   end_col = "Gene.end.PlusKB")
# Annotate BM with 1s where annotation overlaps with the baseline model
list.BM1 <- hits1 %>%
  overlap_annot_hits_w_baseline(BM)
# Export files
annot_basedir1 <- file.path(here::here("data", "ARACNe", "LDSC1/"))
LDSCforRyten::create_annot_file_and_export(list.BM, annot_basedir = annot_basedir)
```

```{r, eval = FALSE}
# Find overlapping regions between genes in each annotation and BM
hits2 <- aracne_ldsc_list2 %>%
  LDSCforRyten::overlap_annot_list(BM_GR, 
                                   seqname_col = "chromosome_name", 
                                   start_col = "Gene.Start.MinusKB", 
                                   end_col = "Gene.end.PlusKB")
# Annotate BM with 1s where annotation overlaps with the baseline model
list.BM2 <- hits2 %>%
  overlap_annot_hits_w_baseline(BM)
# Export files
annot_basedir2 <- file.path(here::here("data", "ARACNe", "LDCS2/"))
LDSCforRyten::create_annot_file_and_export(list.BM, annot_basedir = annot_basedir)
```

# 3. Running LDSC {.tabset}
### Individual regulons
```{r, eval = FALSE}
# Set arguments
args1 <-
  list(
    python_command = "/usr/bin/python2.7",
    annot_basedir = here::here("data", "ARACNe/"),
    annot_name = "LDSC1",
    annot_subcategories = 
      list.dirs(
        path = file.path(here::here("data", "ARACNe", "LDSC1")), 
        recursive = FALSE
        ) %>% 
      basename(),
    baseline_model = "97",
    gwas_df = 
      LDSCforRyten::Create_GWAS_df() %>% 
      dplyr::filter(Original.name %in% c("PD2019.meta5.ex23andMe"))
  )
# Log
print(Sys.time())
print("Start running LDSC...")
# Run
fixed_args <- 
 get_LDSC_fixed_args(Baseline_model = args1$baseline_model)
Calculate_LDscore(
  Command = args1$python_command,
  Annotation_Basedir = args1$annot_basedir,
  Annot_name = args1$annot_name,
  Annotation_Subcategories = args1$annot_subcategories,
  Fixed_Arguments = fixed_args,
  cores = 2
)
# Had to manually move files into individual gene folders as paths weren't quite right
Calculate_H2(
  Command = args1$python_command,
  Annotation_Basedir = args1$annot_basedir,
  Annot_name = args1$annot_name,
  Annotation_Subcategories = args1$annot_subcategories,
  Fixed_Arguments = fixed_args,
  GWAS_df = args1$gwas_df,
  cores = 6
)
# Log
print(Sys.time())
```

### Cummulative frequency
```{r, eval = FALSE}
# Set arguments
args2 <-
  list(
    python_command = "/usr/bin/python2.7",
    annot_basedir = here::here("data", "ARACNe/"),
    annot_name = "LDSC2",
    annot_subcategories = 
      list.dirs(
        path = file.path(here::here("data", "ARACNe", "LDSC2")), 
        recursive = FALSE
        ) %>% 
      basename(),
    baseline_model = "97",
    gwas_df = 
      LDSCforRyten::Create_GWAS_df() %>% 
      dplyr::filter(Original.name %in% c("PD2019.meta5.ex23andMe"))
  )
# Log
print(Sys.time())
print("Start running LDSC...")
# Run
fixed_args <- 
 get_LDSC_fixed_args(Baseline_model = args2$baseline_model)
Calculate_LDscore(
  Command = args2$python_command,
  Annotation_Basedir = args2$annot_basedir,
  Annot_name = args2$annot_name,
  Annotation_Subcategories = args2$annot_subcategories,
  Fixed_Arguments = fixed_args,
  cores = 2
)
# Had to manually move files into individual gene folders as paths weren't quite right
Calculate_H2(
  Command = args2$python_command,
  Annotation_Basedir = args2$annot_basedir,
  Annot_name = args2$annot_name,
  Annotation_Subcategories = args2$annot_subcategories,
  Fixed_Arguments = fixed_args,
  GWAS_df = args2$gwas_df,
  cores = 6
)
# Log
print(Sys.time())
```

# 4. Tidy results {.tabset}
### Individual regulons
```{r, eval = FALSE}
# File paths
file_paths1 <- 
  list.files(
    path = 
      here::here("data", "ARACNe", "LDSC1", "Output"),
    pattern = ".results",
    recursive = T,
    full.names = T
  )
# Filter/collate results
results_w_baseline1  <- 
  file_paths1 %>% 
  lapply(., function(file){
    file_name <- file
    GWAS <- 
      file_name %>% 
      stringr::str_replace("/.*/", "") %>% 
      stringr::str_replace("\\.results", "") %>% 
      stringr::str_replace("_.*", "")
    annot_name <- 
      file_name %>% 
      stringr::str_replace("/.*/", "") %>% 
      stringr::str_replace("\\.results", "") %>% 
      stringr::str_replace(".*baseline_", "")
    readr::read_delim(file = file_name, delim = "\t") %>% 
      dplyr::mutate(
        annot_name = annot_name, 
        GWAS = GWAS
      ) %>% 
      LDSCforRyten::Calculate_enrichment_SE_and_logP(
        ., one_sided = NULL
      ) 
  }) %>% 
  qdapTools::list_df2df() %>% 
  tibble::as_tibble() %>% 
  dplyr::select(-X1)
results1 <-
  results_w_baseline1 %>% 
  dplyr::filter(Category == "L2_0") %>% 
  dplyr::select(-Category)
xlsx1 <- 
  setNames(
    vector(mode = "list", length = 2),
    c("column_descriptions", "results")
  )
xlsx[[2]] <-
  results1 %>% 
  dplyr::group_by(GWAS) %>%
  dplyr::mutate(
    z_score_fdr = 
      p.adjust(
        p = Z_score_P, 
        method = "fdr"
      )
  ) %>% 
  janitor::clean_names() %>% 
  dplyr::select(
    -contains("_se"), -log_p, -z_score_log_p
  )
xlsx1[[1]] <-
  tibble(
    `Column name` = colnames(xlsx[[2]]),
    Description = c(
      "GWAS",
      "Regulon name",
      "Proportion of SNPs accounted for by the annotation within the baseline set of SNPs",
      "Proportion of heritability accounted for by the annotation",
      "Jackknife standard errors for the proportion of heritability. Block jacknife over SNPs with 200 equally sized blocks of adjacent SNPs.",
      "Enrichment = (Proportion of heritability)/(Proportion of SNPs)",
      "Standard error of enrichment",
      "P-value of total enrichment",
      "Regression co-efficient i.e. contribution of annotation after controlling for all other categories in the model",
      "Standard error of coefficient. Estimated using the covariance matrix for coefficient estimates.",
      "Z-score for significance of coefficient",
      "P-value for coefficient computed from z-score",
      "FDR-adjusted coefficient p-value (adjusted by number of percentiles and selection metrics)"
    )
  )
```

```{r, eval = FALSE}
# Save files 
write_delim(
  results_w_baseline1, 
  file = here::here("data", "ARACNe", "LDSC1", "ldsc1_summary_w_baseline_annot.txt"),
  delim = "\t"
)  
write_delim(
  results1, 
  file = here::here("data", "ARACNe", "LDSC1", "ldsc1_summary.txt"),
  delim = "\t"
) 
openxlsx::write.xlsx(
  xlsx1,
  file = here::here("data", "ARACNe", "LDSC1", "ldsc1_table.xlsx"),
  rowNames = FALSE,
  headerStyle = openxlsx::createStyle(textDecoration = "BOLD"),
  firstRow = TRUE,
  append = TRUE
)
```

```{r}
# Set arguments
args1 <- 
  list(
    results_dir = here::here("data", "ARACNe", "LDSC1")
  )
# Load files
NSL1 <- 
  read_excel(
    file.path(args1$results_dir, "ldsc1_table.xlsx"),
    sheet = "results"
  )
# Main table
NSL_sig1 <- 
  NSL1 %>% 
  mutate(
    FDR_Pvalue = 
      ifelse(
        z_score_fdr < 0.05, 
        "FDR p-value < 0.05", 
        "Not significant, FDR p-value >=0.05"
        ),
    Coefficient.Lower.SE = coefficient - coefficient_std_error,
    Coefficient.Upper.SE = coefficient + coefficient_std_error,
    regulons = 
      annot_name %>% 
      factor()
    )
NSL_sig1 %>%
  dplyr::mutate(enrichment_fdr = p.adjust(enrichment_p, method = "fdr")) %>%
  dplyr::select(annot_name, enrichment_p, enrichment_fdr, z_score_p, z_score_fdr)
```

### Cummulative frequency
```{r, eval = FALSE}
# File paths
file_paths2 <- 
  list.files(
    path = 
      here::here("data", "ARACNe", "LDSC2", "Output"),
    pattern = ".results",
    recursive = T,
    full.names = T
  )
# Filter/collate results
results_w_baseline2  <- 
  file_paths2 %>% 
  lapply(., function(file){
    file_name <- file
    GWAS <- 
      file_name %>% 
      stringr::str_replace("/.*/", "") %>% 
      stringr::str_replace("\\.results", "") %>% 
      stringr::str_replace("_.*", "")
    annot_name <- 
      file_name %>% 
      stringr::str_replace("/.*/", "") %>% 
      stringr::str_replace("\\.results", "") %>% 
      stringr::str_replace(".*baseline_", "")
    readr::read_delim(file = file_name, delim = "\t") %>% 
      dplyr::mutate(
        annot_name = annot_name, 
        GWAS = GWAS
      ) %>% 
      LDSCforRyten::Calculate_enrichment_SE_and_logP(
        ., one_sided = NULL
      ) 
  }) %>% 
  qdapTools::list_df2df() %>% 
  tibble::as_tibble() %>% 
  dplyr::select(-X1)
results2 <-
  results_w_baseline2 %>% 
  dplyr::filter(Category == "L2_0") %>% 
  dplyr::select(-Category)
xlsx2 <- 
  setNames(
    vector(mode = "list", length = 2),
    c("column_descriptions", "results")
  )
xlsx2[[2]] <-
  results2 %>% 
  dplyr::group_by(GWAS) %>%
  dplyr::mutate(
    z_score_fdr = 
      p.adjust(
        p = Z_score_P, 
        method = "fdr"
      )
  ) %>% 
  janitor::clean_names() %>% 
  dplyr::select(
    -contains("_se"), -log_p, -z_score_log_p
  )
xlsx2[[1]] <-
  tibble(
    `Column name` = colnames(xlsx[[2]]),
    Description = c(
      "GWAS",
      "Regulon cummulative frequency",
      "Proportion of SNPs accounted for by the annotation within the baseline set of SNPs",
      "Proportion of heritability accounted for by the annotation",
      "Jackknife standard errors for the proportion of heritability. Block jacknife over SNPs with 200 equally sized blocks of adjacent SNPs.",
      "Enrichment = (Proportion of heritability)/(Proportion of SNPs)",
      "Standard error of enrichment",
      "P-value of total enrichment",
      "Regression co-efficient i.e. contribution of annotation after controlling for all other categories in the model",
      "Standard error of coefficient. Estimated using the covariance matrix for coefficient estimates.",
      "Z-score for significance of coefficient",
      "P-value for coefficient computed from z-score",
      "FDR-adjusted coefficient p-value (adjusted by number of percentiles and selection metrics)"
    )
  )
```

```{r, eval = FALSE}
# Save files 
write_delim(
  results_w_baseline2, 
  file = here::here("data", "ARACNe", "LDSC2", "ldsc2_summary_w_baseline_annot.txt"),
  delim = "\t"
)  
write_delim(
  results2, 
  file = here::here("data", "ARACNe", "LDSC2", "ldsc2_summary.txt"),
  delim = "\t"
) 
openxlsx::write.xlsx(
  xlsx2,
  file = here::here("data", "ARACNe", "LDSC2", "ldsc2_table.xlsx"),
  rowNames = FALSE,
  headerStyle = openxlsx::createStyle(textDecoration = "BOLD"),
  firstRow = TRUE,
  append = TRUE
)
```

```{r}
# Set arguments
args2 <- 
  list(
    results_dir = here::here("data", "ARACNe", "LDSC2")
  )
# Load files
NSL2 <- 
  read_excel(
    file.path(args2$results_dir, "ldsc2_table.xlsx"),
    sheet = "results"
  )
# Main table
NSL_sig2 <- 
  NSL2 %>% 
  mutate(
    FDR_Pvalue = 
      ifelse(
        z_score_fdr < 0.05, 
        "FDR p-value < 0.05", 
        "Not significant, FDR p-value >=0.05"
        ),
    Coefficient.Lower.SE = coefficient - coefficient_std_error,
    Coefficient.Upper.SE = coefficient + coefficient_std_error,
    regulons = 
      annot_name %>% 
      factor()
    )
NSL_sig2 %>%
  dplyr::mutate(enrichment_fdr = p.adjust(enrichment_p, method = "fdr")) %>%
  dplyr::select(annot_name, enrichment_p, enrichment_fdr, z_score_p, z_score_fdr)
```
