---
title: "Tissue and cell type specificity analysis"
author: "Amy Hicks"
date: "29/09/2022"
output: 
  html_document:
    code_folding: hide
    highlight: kate
    df_print: paged
    toc: true
    toc_float: true
---

```{r setup, include=FALSE}
library(DT)
library(readxl)
library(ggplot2)
library(ggpubr)
library(tidyverse)
library(stringr)
library(here)
knitr::opts_chunk$set(echo = TRUE)
```

This markdown examines the tissue- and cell-type specificity of NSL complex genes using single-cell nuclear RNA-sequencing data. For the latter, the `MarkerGenes` repository was cloned from [Regina's github](https://github.com/RHReynolds/MarkerGenes.git) and run locally.  

# 1. Cell type 
## 1.1. Load data
**Substantia nigra** data from [Argwal et al., 2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7442652/) and human **middle temporal gyrus** (MTG) data from the [Allen Brain Atlas](https://celltypes.brain-map.org/rnaseq/human/mtg) were used.
```{r}
# Load specificity matrices
load(file.path(here::here("data/MarkerGenes_source/AIBS2018_MTG.rda")))
load(file.path(here::here("data/MarkerGenes_source/Agarwal2020_SNIG.rda")))
# Import lists of genes of interest
NSL_file <- read.csv(file.path(here::here("data/Gene_lists/NSL_genes.csv")))
names(NSL_file)[names(NSL_file)=="Symbol"] <- "Gene_symbol"
NSL_genes <- as.character(NSL_file$Gene_symbol)
```

## 1.2. Search genes
```{r}
# Query gene list
source(file.path(here::here("scripts/MarkerGenes_source/query_gene_ctd.R")))
specificity <- query_gene_ctd(genes = NSL_genes,
                              ctd_Agarwal2020_SNIG, ctd_AIBS2018,
                              celltypeLevel = 1,
                              median_included = FALSE,
                              genelistSpecies = "human",
                              ctdSpecies = "human")
specificity$CellType <- as.character(specificity$CellType)
```

## 1.3. Plot results
```{r}
# Rename cell types
specificity$CellType[specificity$CellType=="ODC"] <- "Oligodendrocytes"
specificity$CellType[specificity$CellType=="Oligodendrocyte"] <- "Oligodendrocytes"
specificity$CellType[specificity$CellType=="Astrocyte"] <- "Astrocytes"
specificity$CellType[specificity$CellType=="GABA"] <- "GABAergic neurons"
specificity$CellType[specificity$CellType=="GABAergic"] <- "GABAergic neurons"
specificity$CellType[specificity$CellType=="Endothelial cell"] <- "Endothelial cells"
specificity$CellType[specificity$CellType=="Endothelial"] <- "Endothelial cells"
specificity$CellType[specificity$CellType=="DaNs"] <- "Dopaminergic neurons"
specificity$CellType[specificity$CellType=="Glutamatergic"] <- "Glutamatergic neurons"
specificity$CellType[specificity$CellType=="OPC"] <- "Oligodendrocyte precursor cells"
specificity$Study[specificity$Study=="ctd_Agarwal2020_SNIG"] <- "Substantia nigra"
specificity$Study[specificity$Study=="ctd_AIBS2018"] <- "Medial temporal gyrus"
# Plot results
source(file.path(here::here("scripts/MarkerGenes_source/reorder_within.R")))
specificity %>% 
  ggplot(aes(x = reorder_within(x = CellType,
                                by = Specificity,
                                within = Study,
                                fun = median,
                                desc = TRUE),
             y = Specificity)) +
  geom_col() +
  scale_x_reordered() +
  facet_wrap(~ Study, scales = "free_x") +
  labs(x = "Cell type") +
  theme(axis.text.x = element_text(angle = 90, size = 7, hjust = 1, vjust = 0.5))
```

```{r}
# Plot just substantia nigra neurons
specificity %>% 
  dplyr::filter(Study == "Substantia nigra") %>%
  dplyr::filter(CellType %in% c("Dopaminergic neurons", "GABAergic neurons", "Glutamatergic neurons")) %>%
  ggplot(aes(x = reorder_within(x = Gene,
                                by = Specificity,
                                within = Study,
                                fun = median,
                                desc = TRUE),
             y = Specificity)) +
  geom_col(fill = "goldenrod3") +
  scale_x_reordered() +
  facet_wrap(~ CellType, scales = "free_x") +
  labs(x = "Gene") +
  theme(axis.text.x = element_text(size = 7))
```

```{r}
# Plot just MTG neurons
specificity %>% 
  dplyr::filter(Study == "Medial temporal gyrus") %>%
  dplyr::filter(CellType %in% c("Dopaminergic neurons", "GABAergic neurons", "Glutamatergic neurons")) %>%
  ggplot(aes(x = reorder_within(x = Gene,
                                by = Specificity,
                                within = Study,
                                fun = median,
                                desc = TRUE),
             y = Specificity)) +
  geom_col(fill = "goldenrod3") +
  scale_x_reordered() +
  facet_wrap(~ CellType, scales = "free_x") +
  labs(x = "Gene") +
  theme(axis.text.x = element_text(size = 7))
```

# 2. Tissue specificty
## 2.1. Load data
Median transcripts per million (TPM) values for the NSL genes across **13 brain regions** were taken from [GTEx](https://www.gtexportal.org/home/). 
```{r}
# Open file
NSL_gtex_tissueexp <- read_excel(file.path(here::here("data/NSL_gtex_tissueexp.xlsx")))
```

## 2.2. Plot results
```{r}
# Format for plot
NSL_gtex_tissueexp$Median_TPM <- as.numeric(NSL_gtex_tissueexp$Median_TPM)
NSL_gtex_tissueexp <- dplyr::arrange(NSL_gtex_tissueexp, Tissue)
NSL_gtex_tissueexp$position <- nrow(NSL_gtex_tissueexp):1
# Plot
ggplot(NSL_gtex_tissueexp, aes(Gene, reorder(Tissue, position))) +
  geom_tile(aes(fill = Median_TPM)) +
  scale_fill_gradient(low = "white", high = "goldenrod3") +
  labs(y = "Tissue")
```